Sage/carbon
# chore(detail): remove classic theme

removes classic theme stories and old documentation. This is replaced with CSF and MDX examples of
the detail component.

fixes FE-3137

### Proposed behaviour
Removal of `classic theme` stories from the Detail component. 
The documentation folder has been deleted.
StoriesOf story has been refactored to be in CSF.
An MDX example for Detail has been created.

### Current behaviour
Detail has a `classic theme` story.

### Checklist



### Additional context
N/A

### Testing instructions
- `git checkout FE-3137-remove_classic_theme_detail`
- Start storybook by using `npm start`
- Navigate to the Detail story in Storybook
- No `classic` story should be present.
- Detail should have an MDX example that documents the component.
